fdbclient/FDBTypes.h

```diff
@@ -277,7 +277,19 @@ struct KeyRangeRef {
 
 	template <class Ar>
 	force_inline void serialize(Ar& ar) {
-		serializer(ar, const_cast<KeyRef&>(begin), const_cast<KeyRef&>(end));
+		if (!ar.isDeserializing && equalsKeyAfter(begin, end)) {
+			KeyRef hold = begin;
+			const_cast<KeyRef&>(begin) = KeyRef();
```

Comment:
You don't need to change begin - you can pass a local variable to `serializer` instead of begin.

Reply:
When I tried to pass a local variable I got compiler errors, I could not figure out why, any clues?

Reply:
Interesting. I'll take a look at that. In that case this comment shouldn't block your PR.

Reply:
This patch seems to compile for me:
```
$ git diff
diff --git a/fdbclient/CommitTransaction.h b/fdbclient/CommitTransaction.h
index 6349b9bc..c7dab536 100644
--- a/fdbclient/CommitTransaction.h
+++ b/fdbclient/CommitTransaction.h
@@ -97,10 +97,8 @@ struct MutationRef {
        template <class Ar>
        void serialize( Ar& ar ) {
                if (!ar.isDeserializing && type == ClearRange && equalsKeyAfter(param1, param2)) {
-                       StringRef hold = param1;
-                       param1 = StringRef();
-                       serializer(ar, type, param2, param1);
-                       param1 = hold;
+                       StringRef empty;
+                       serializer(ar, type, param2, empty);
                } else {
                        serializer(ar, type, param1, param2);
                }
diff --git a/fdbclient/FDBTypes.h b/fdbclient/FDBTypes.h
index d3e22dd1..a32305ac 100644
--- a/fdbclient/FDBTypes.h
+++ b/fdbclient/FDBTypes.h
@@ -278,10 +278,8 @@ struct KeyRangeRef {
        template <class Ar>
        force_inline void serialize(Ar& ar) {
                if (!ar.isDeserializing && equalsKeyAfter(begin, end)) {
-                       KeyRef hold = begin;
-                       const_cast<KeyRef&>(begin) = KeyRef();
-                       serializer(ar, const_cast<KeyRef&>(end), const_cast<KeyRef&>(begin));
-                       const_cast<KeyRef&>(begin) = hold;
+                       KeyRef empty;
+                       serializer(ar, const_cast<KeyRef&>(end), empty);
                } else {
                        serializer(ar, const_cast<KeyRef&>(begin), const_cast<KeyRef&>(end));
                }
```